ipa-core/src/bin/report_collector.rs

```
@@ -171,25 +203,150 @@ fn gen_inputs(
+        let file = File::open(path).unwrap_or_else(|e| panic!("unable to open file {path:?}. {e}"));
+        let reader = BufReader::new(file);
+        for line in reader.lines() {
+            let line = line.unwrap();
+            let encrypted_report_bytes = hex::decode(line.trim()).unwrap();
+            buffers[i].put_u16_le(encrypted_report_bytes.len().try_into().unwrap());
+            buffers[i].put_slice(encrypted_report_bytes.as_slice());
         }
-        (IpaSecurityModel::Malicious, IpaQueryStyle::Oprf) => {
-            panic!("OPRF for malicious is not implemented as yet")
+    }
+    // Panic if input sizes are not the same
+    assert!(buffers[0].len() == buffers[1].len());
+    assert!(buffers[1].len() == buffers[2].len());
+
+    // without loss of generality, set query length to length of first input size
+    let query_size = buffers[0].len();
+
+    let query_config = QueryConfig {
+        size: QuerySize::try_from(query_size).unwrap(),
+        field_type: FieldType::Fp32BitPrime,
+        query_type,
+    };
+
+    let query_id = helper_clients[0]
+        .create_query(query_config)
+        .await
+        .expect("Unable to create query!");
+
+    let inputs = buffers.map(BodyStream::from);
+    tracing::info!("Starting query for OPRF");
+    let actual = match query_style {
+        IpaQueryStyle::Oprf => {
+            // the value for histogram values (BA32) must be kept in sync with the server-side
+            // implementation, otherwise a runtime reconstruct error will be generated.
+            // see ipa-core/src/query/executor.rs
+            run_query_and_validate::<BA32>(
+                inputs,
+                query_size,
+                helper_clients,
+                query_id,
+                ipa_query_config,
+            )
+            .await
         }
     };
 
+    if let Some(ref path) = args.output_file {
+        let _ = write_ipa_output_file(path, &actual);
```

this just swallows the errors, you should use ?` instead to propagate them